### PR TITLE
Feat 14 get user

### PR DESCRIPTION
**Pull request by YaronAvraham93**

### CODE REVIEW

File: budget-app/src/components/molecules/Activity/ActivitesChart.tsx [hunk 1]
```diff
@@ -18,14 +17,36 @@ const Container = styled.div(
 );
 const ActivitesChart: React.FC = () => {
   const { transactions } = useContext(TransactionContext);
+  const helperIncome: any = {};
```
Comment (IsaacM890) [inline budget-app/src/components/molecules/Activity/ActivitesChart.tsx]: should ne dynamic function 
export it to helpers/utils folder

File: budget-app/src/components/organisms/Sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -36,18 +37,22 @@ const NavbarList = styled.ul`
...
-    <NavbarList>
-      {navbarItem.map(({ icon, title, route, id }) => (
-        <Link to={route}>
-          <NavbarItem icon={icon} title={title} key={id} />
-        </Link>
-      ))}
-    </NavbarList>
-  </Container>
-);
+const Sidebar: React.FC = () => {
```
Comment (IsaacM890) [inline budget-app/src/components/organisms/Sidebar/Sidebar.tsx]: maybe "NavBar" is better name

File: budget-app/src/components/organisms/UserInfro/UserInfro.tsx [hunk 3]
```diff
@@ -29,20 +30,18 @@ const PaymentListWrapper = styled.div`
...
-        lastName="Avraham"
-        email="yaronavraham600@gmail.com"
-      />
-    </UserWrapper>
-    <PaymentListWrapper>
-      <PaymentHistory />
-    </PaymentListWrapper>
-  </Container>
-);
+const UserInfro: React.FC = () => {
```
Comment (IsaacM890) [inline budget-app/src/components/organisms/UserInfro/UserInfro.tsx]: typo == > "UserInfo" not "UserInfro"

**Review by IsaacM890 — CHANGES_REQUESTED** [review]

File: budget-app/src/contexts/userReducer.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { IUserInitialState, IAction } from '../models/index';
+
+const userReducer = (state: IUserInitialState, action: IAction) => {
+  switch (action.type) {
+    case 'SET_USER':
```
Comment (almogbhl) [inline budget-app/src/contexts/userReducer.ts]: constant

File: budget-app/src/helpers/totalAmountByPaymentType.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { ITransaction } from '../models/index';
+
+const getTotalAmountByPaymentType = (transactions: ITransaction[], Type: string) => {
+  const helper: any = {};
+  // const helperExpenses: any = {};
```
Comment (almogbhl) [inline budget-app/src/helpers/totalAmountByPaymentType.ts]: rm unused code

File: budget-app/src/helpers/totalAmountByPaymentType.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  transactions.forEach((transaction: ITransaction) => {
+    const date = new Date(transaction.date).toLocaleString('default', { month: 'short' });
+
+    if (transaction.paymentType === Type) {
+      if (!helper[date]) {
+        helper[date] = transaction.amount;
+      } else {
+        helper[date] = +transaction.amount;
```
Comment (almogbhl) [inline budget-app/src/helpers/totalAmountByPaymentType.ts]: +=

File: budget-app/src/helpers/totalAmountByPaymentType.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { ITransaction } from '../models/index';
+
+const getTotalAmountByPaymentType = (transactions: ITransaction[], Type: string) => {
+  const helper: any = {};
+  // const helperExpenses: any = {};
+  transactions.forEach((transaction: ITransaction) => {
+    const date = new Date(transaction.date).toLocaleString('default', { month: 'short' });
+
+    if (transaction.paymentType === Type) {
```
Comment (almogbhl) [inline budget-app/src/helpers/totalAmountByPaymentType.ts:8]: u wish to do something else if the type is income or expense

File: budget-app/src/models/index.tsx [hunk 8]
```diff
@@ -23,11 +23,11 @@ export interface ICurrentBalanceProps {
   text?: string;
 }
 export interface ICreditCardProps {
-  last4Digits: number;
+  last4Digits: Number;
```
Comment (almogbhl) [inline budget-app/src/models/index.tsx]: number

**Review by almogbhl — CHANGES_REQUESTED** [review]

File: budget-app/src/models/index.tsx [hunk 9]
```diff
@@ -48,7 +48,7 @@ export interface ITransaction {
   date: string;
   paymentType: string;
   paymentMethod: string;
-  amount: Number;
+  amount: number;
```
Comment (almogbhl) [inline budget-app/src/models/index.tsx:51]: string

**Review by almogbhl — APPROVED** [review]

**Pull request merged**